Composer/packages/client/src/pages/design/DesignPage.tsx

```diff
@@ -201,6 +203,11 @@ const DesignPage: React.FC<RouteComponentProps<{ dialogId: string; projectId: st
   useEffect(() => {
     if (location && props.dialogId && props.projectId) {
       const { dialogId, projectId } = props;
+
+      // TODO: swap to the commented-out block once we're working on skills for real
+      // let { skillId } = props;
+      // if (skillId == null) skillId = projectId;
```

Comment:
```
 const { location, dialogId, projectId = '' } = props;
```

So i dont see references to the skillId being set from the props. Essentially, with this current Design Page how would skills be supported. It seems like the schema, shellApi are all going to be only in the context of the rootBot. Seems like it should be changed to

```javascript
const { location, dialogId, projectId = '', skillId } = props;
const currentProjectId = projectId || skillId

Now all the useRecoilValues would use something like useRecoilValue(schemasState(currentProjectId))
```

Reply:
I left skillId out of the props on purpose because, at this stage, we don't need it yet. This is a minimal change to get the new ProjectTree design in without overcomplicating things. The older branch I'm pulling stuff from did read skillId, just like you're suggesting.